Index: src/org/omegat/util/StaticUtils.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>windows-1251
===================================================================
--- src/org/omegat/util/StaticUtils.java	(revision a09dcf0d14784f0e6a1b5fc3f015ab82007eb132)
+++ src/org/omegat/util/StaticUtils.java	(revision b751e5390b3ddeb4a69bdcefed58613aeb10ecbb)
@@ -179,13 +179,13 @@
      * @return a string containing the tags
      */
     public static String buildTagListForRemove(String str) {
-        String res = "";
+        StringBuilder result = new StringBuilder();
         Pattern placeholderPattern = PatternConsts.OMEGAT_TAG;
         Matcher placeholderMatcher = placeholderPattern.matcher(str);
         while (placeholderMatcher.find()) {
-            res += placeholderMatcher.group(0);
+            result.append(placeholderMatcher.group(0));
         }
-        return res;
+        return result.toString();
     }
     
     /**
@@ -307,6 +307,7 @@
     /**
      * Returns a list of all files under the root directory by absolute path.
      */
+    //todo don't pass files collection, but return it instead
     public static void buildFileList(List<String> lst, File rootDir, boolean recursive) {
         internalBuildFileList(lst, rootDir, recursive);
 
